pkg/signer/request-signature-v4.go

```diff
@@ -289,7 +289,7 @@ func signV4(req http.Request, accessKeyID, secretAccessKey, sessionToken, locati
 			req.Header.Add("X-Amz-Trailer", strings.ToLower(k))
 		}
 
-		req.TransferEncoding = []string{"aws-chunked"}
+		req.Header.Set("Content-Encoding", "aws-chunked")
```

Comment:
are you sure that this won't break anything?

Reply:
@harshavardhana It is the correct value. 

From https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html

> [Y]ou must include the following headers:
> [...] Content-Encoding: Set the value to `aws-chunked`.

Reply:
This basically broke if you remember last time? @klauspost

Reply:
@harshavardhana No. The only thing I remember was that what we tried didn't make any difference (since this is the wrong field). Please refresh my memory.